common/src/main/java/com/wynntils/features/EmeraldPouchHotkeyFeature.java

```diff
@@ -0,0 +1,197 @@
+                        // Check to make sure we don't have a bunch of zero balances - if we do, open largest capacity
+                        if (!alreadyHasNonEmpty) {
+                            Map.Entry<Integer, Pair<ItemStack, Pair<Integer, Integer>>> largest = null;
+                            for (Map.Entry<Integer, Pair<ItemStack, Pair<Integer, Integer>>> entry :
+                                    emeraldPouches.entrySet()) {
+                                if (largest == null || entry.getValue().b.b > largest.getValue().b.b) {
+                                    largest = entry;
+                                }
+                            }
+                            slotNumber = largest.getKey();
+                            slotStack = largest.getValue().a;
+                        }
+
+                        // Now, we know we have 1 used and 1+ empty pouches - open the used one we saved from before
+                        if (slotNumber == -1) {
+                            slotNumber = usedPouch;
+                        }
+                        if (slotStack == null) {
+                            slotStack = pouchStack;
+                        }
+                        if (slotNumber < 9) {
+                            // sendPacket uses raw slot numbers, we need to remap the hotbar
+                            slotNumber += 36;
+                        }
+                        changedSlots.putIfAbsent(slotNumber, slotStack);
+                        McUtils.player()
+                                .connection
+                                .send(new ServerboundContainerClickPacket(
+                                        player.inventoryMenu.containerId,
+                                        player.inventoryMenu.getStateId(),
+                                        slotNumber,
+                                        1,
+                                        ClickType.PICKUP,
+                                        ItemStack.EMPTY,
+                                        changedSlots));
+                    }
+                    case 0 -> McUtils.sendMessageToClient(
+                            new TranslatableComponent("feature.wynntils.emeraldPouchKeybind.noPouch")
+                                    .withStyle(ChatFormatting.DARK_RED));
+
+                    case 1 -> {
+                        int slotNumber = emeraldPouches
+                                .entrySet()
+                                .iterator()
+                                .next()
+                                .getKey(); // Just get the first value in the HashMap since we only have one value
+
+                        if (slotNumber < 9) {
+                            // sendPacket uses raw slot numbers, we need to remap the hotbar
+                            slotNumber += 36;
+                        }
+                        changedSlots.putIfAbsent(
```

Comment:
`changedSlots` declaration could be moved here for readability, or even inlined entirely